olp-cpp-sdk-dataservice-read/src/generated/api/PlatformApi.h

```diff
@@ -39,37 +40,20 @@ namespace read {
 class PlatformApi {
  public:
   using ApisResponse = client::ApiResponse<model::Apis, client::ApiError>;
-  using ApisCallback = std::function<void(ApisResponse)>;
-
-  /**
-   * @deprecated
-   * @brief Call to lookup platform base urls.
-   * @param client Instance of OlpClient used to make REST request.
-   * @param service Name of the service.
-   * @param serviceVersion Version of the service.
-   * @param A callback function to invoke with the collection of Api services
-   * that match the parameters.
-   *
-   * @return The cancellation token.
-   */
-  static client::CancellationToken GetApis(
-      std::shared_ptr<client::OlpClient> client, const std::string& service,
-      const std::string& serviceVersion, const ApisCallback& callback);
 
   /**
    * @brief Call to lookup platform base urls.
    * @param client Instance of OlpClient used to make REST request.
    * @param service Name of the service.
    * @param service_version Version of the service.
-   * @param A callback function to invoke with the collection of Api services
-   * that match the parameters.
+   * @param context A CancellationContext, which can be used to cancel any pending request.
    *
-   * @return The cancellation token.
+   * @return The Apis response.
    */
-  static client::CancellationToken GetApis(const client::OlpClient& client,
-                                           const std::string& service,
-                                           const std::string& service_version,
-                                           const ApisCallback& callback);
+  static ApisResponse GetApis(const client::OlpClient& client,
+                              const std::string& service,
+                              const std::string& service_version,
+                              const client::CancellationContext& context);
```

Comment:
no need for const reference, we can pass it by value

Reply:
I presume we shouldn't as the initial call is a sync call. We can copy it internally. I don't see a point in copying if you don't need to.